.github/workflows/ccpp.yml

```diff
@@ -86,7 +86,6 @@ jobs:
           python3 -m isort --check .
       - name: mypy checks
         run: |
-          python3 -m pip install mypy types-setuptools --upgrade # TO REMOVE
```

Comment:
The current workflow upgrades `mypy` at every run, i.e. if the version of `mypy` on `ghcr.io/fenics/test-env:current-openmpi` is old it will get upgraded. 

From https://github.com/FEniCS/dolfinx/actions/workflows/docker-dev-test-env.yml I see that the  `docker-dev-test-env.yml` is manually triggered once a month or so, so I do not expect the version of `mypy` will ever be too old.

Reply:
The workflows should not be using `pip` to install/upgrade dependencies or related tools. Everything should be in the appropriate `Dockerfile`.

Sometimes it's necessary to install something in a workflow to get the tests to pass and the PR through. The images should then be rebuilt and a cleanup PR created. 

It's good practice to add a note e.g. `# TO REMOVE` as it's easy to forget.

Reply:
The only remaining `pip install` in these three workflows is https://github.com/FEniCS/dolfinx/blame/ef7255b655e96c88ef9556efa982df1c00987cf7/.github/workflows/redhat.yml#L27
which was just added 3 days ago. Do you want me to get rid of that as well?

Reply:
> The current workflow upgrades `mypy` at every run, i.e. if the version of `mypy` on `ghcr.io/fenics/test-env:current-openmpi` is old it will get upgraded. 
> 
> 
> 
> From https://github.com/FEniCS/dolfinx/actions/workflows/docker-dev-test-env.yml I see that the  `docker-dev-test-env.yml` workflow is manually triggered once a month or so, so I do not expect the version of `mypy` will ever be too old.

You're welcome to try. Would be interesting to see if redhat's pip is up to the task. If not we need to adjust the Dockerfile.